solr/core/src/java/org/apache/solr/handler/admin/SplitOp.java

```diff
@@ -385,16 +385,8 @@ static String toSplitString(Collection<DocRouter.Range> splits) throws Exception
 
       // compare to current prefix bucket and see if this new term shares the same prefix
       if (term != null && term.length >= currPrefix.length && currPrefix.length > 0) {
-        int i = 0;
-        for (; i < currPrefix.length; i++) {
-          if (currPrefix.bytes[i] != term.bytes[term.offset + i]) {
-            break;
-          }
-        }
-
-        if (i == currPrefix.length) {
-          // prefix was the same (common-case fast path)
-          // int count = termsEnum.docFreq();
+        BytesRef termPrefix = new BytesRef(term.bytes, term.offset, currPrefix.length);
```

Comment:
Found a better choice here that doesn't involve  object creation: StringHelper.startsWith()